raiden/api/python.py

```diff
@@ -768,12 +768,12 @@ def transfer_async(
                 raise InvalidSecretOrSecretHash('secret_hash is not an hexadecimal string.')
             secret_hash = to_bytes(hexstr=secret_hash)
 
-        if secret is None and secret_hash is not None:
-            raise InvalidSecretOrSecretHash('secret_hash without a secret is not supported yet.')
-
         if secret is not None and secret_hash is not None and secret_hash != sha3(secret):
             raise InvalidSecretOrSecretHash('provided secret and secret_hash do not match.')
 
+        if secret is None and secret_hash is not None:
+            pass
```

Comment:
What changes if you just remove these two lines?

Reply:
Nothing.
I assume that it is being removed automatically at run time.

Only reason for these lines is clarify - without it the reader may think that this case was forgotten. 

Will remove the lines.

Reply:
IF you want to keep them then at least leave a comment as to why/what this is.

Reply:
Removed and left a comment